test/byContract/registries/resources.test.ts

```diff
@@ -0,0 +1,224 @@
+    ReverseRegistrar__factory,
+    Workerpool,
+    WorkerpoolRegistry,
+    WorkerpoolRegistry__factory,
+    Workerpool__factory,
+} from '../../../typechain';
+import { MULTIADDR_BYTES } from '../../../utils/constants';
+import { getIexecAccounts } from '../../../utils/poco-tools';
+
+describe('Resources', () => {
+    let proxyAddress: string;
+    let iexecPoco: IexecInterfaceNative;
+    let [appProvider, datasetProvider, scheduler, anyone]: SignerWithAddress[] = [];
+
+    let ensRegistry: ENSRegistry;
+    let appRegistry: AppRegistry;
+    let datasetRegistry: DatasetRegistry;
+    let workerpoolRegistry: WorkerpoolRegistry;
+
+    let app: App;
+    let dataset: Dataset;
+    let workerpool: Workerpool;
+
+    beforeEach(async () => {
+        proxyAddress = await loadHardhatFixtureDeployment();
+        await loadFixture(initFixture);
+    });
+
+    async function initFixture() {
+        ({ appProvider, datasetProvider, scheduler, anyone } = await getIexecAccounts());
+        const ensRegistryAddress = (await deployments.get('ENSRegistry')).address;
+        ensRegistry = ENSRegistry__factory.connect(ensRegistryAddress, anyone);
+
+        iexecPoco = IexecInterfaceNative__factory.connect(proxyAddress, anyone);
+
+        appRegistry = AppRegistry__factory.connect(await iexecPoco.appregistry(), anyone);
+        datasetRegistry = DatasetRegistry__factory.connect(
+            await iexecPoco.datasetregistry(),
+            anyone,
+        );
+        workerpoolRegistry = WorkerpoolRegistry__factory.connect(
+            await iexecPoco.workerpoolregistry(),
+            anyone,
+        );
+    }
+
+    describe('App', () => {
+        const createAppArgs = [
+            `App`,
+            'DOCKER',
+            MULTIADDR_BYTES,
+            ethers.utils.id(`Content of my app`),
+            '0x1234',
+        ] as [string, string, BytesLike, BytesLike, BytesLike];
+
+        beforeEach(async () => {
+            const appAddress = await appRegistry.callStatic.createApp(
+                appProvider.address,
+                ...createAppArgs,
+            );
+            await appRegistry
+                .createApp(appProvider.address, ...createAppArgs)
+                .then((tx) => tx.wait());
+            app = App__factory.connect(appAddress, anyone);
+        });
+
+        it('Should create an app and verify its details', async () => {
+            expect(await app.registry()).to.equal(appRegistry.address);
+            expect(await app.owner()).to.equal(appProvider.address);
+            expect(await app.m_appName()).to.equal(createAppArgs[0]);
+            expect(await app.m_appType()).to.equal(createAppArgs[1]);
+            expect(await app.m_appMultiaddr()).to.equal(createAppArgs[2]);
+            expect(await app.m_appChecksum()).to.equal(createAppArgs[3]);
+            expect(await app.m_appMREnclave()).to.equal(createAppArgs[4]);
+        });
+
+        it('Should set the ENS name for the app', async () => {
```

Comment:
```suggestion
        it('Should set name for the ENS reverse registration of the <app/dataset/pool>', async () => {
```
Here and in registries I would suggest